core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -33,10 +33,17 @@ var (
 	ErrHeadNotAvailable   = fmt.Errorf("head not available")
 	ErrBlockLimitExceeded = fmt.Errorf("block limit exceeded")
 
-	// AllowedLogsPerUpkeep is the maximum number of logs allowed per upkeep every single call.
-	AllowedLogsPerUpkeep = 5
-	// MaxPayloads is the maximum number of payloads to return per call.
-	MaxPayloads = 100
+	// MaxPayloadsPerUpkeep is the maximum number of logs allowed per upkeep every single call to the log provider.
+	MaxPayloadsPerUpkeep = 5
+	// MaxPayloads is the maximum number of payloads to return per call for all upkeeps.
+	//
+	// using this limit, we can calculate the maximum number of logs to return per upkeep:
+	// MaxPayloads / logsPerUpkeep = number of upkeeps that we can process each second
+	// 1. ideal case (a single log per upkeep)
+	// MaxPayloads / 1 = 500
+	// 2. worst case (all logs are for the same upkeep)
+	// MaxPayloads / MaxPayloadsPerUpkeep = 100
+	MaxPayloads = 500
```

Comment:
We don't know whether attached RPCs to NOPs can handle these number of payloads per second yet